geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,55 +8,234 @@
+    val dsParams = datastoreParams
+    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+    if (ds == null) {
+      BadRequest(reason = "Could not load data store using the provided parameters.")
     } else {
-      val (returnBytes, filename) = abs.get(id)
-      if (returnBytes == null) {
-        NotFound(reason = s"Unknown ID $id")
-      } else {
-        contentType = "application/octet-stream"
-        response.setHeader("Content-Disposition", "attachment; filename=" + filename)
-
-        Ok(returnBytes)
+      val alias = params("alias")
+      val prefix = keyFor(alias)
+      val toPersist = dsParams.map { case (k, v) => keyFor(alias, k) -> v }
+      try {
+        persistence.removeAll(persistence.keys(prefix).toSeq)
+        persistence.persistAll(toPersist)
+        blobStores.put(alias, new AccumuloBlobStore(ds))
+        Ok()
+      } catch {
+        case e: Exception => handleError(s"Error persisting data store '$alias':", e)
       }
     }
   }
+
+  /**
+    * Retrieve an existing data store
+    */
+  get("/ds/:alias/?") {
+    try {
+      getPersistedDataStore(params("alias"))
+    } catch {
+      case e: Exception => handleError(s"Error reading data store:", e)
+    }
+  }
+
+  /**
+    * Remove the reference to an existing data store
+    */
+  delete("/ds/:alias/?") {
+    val alias = params("alias")
+    val prefix = keyFor(alias)
+    try {
+      persistence.removeAll(persistence.keys(prefix).toSeq)
+      Ok()
+    } catch {
+      case e: Exception => handleError(s"Error removing data store '$alias':", e)
+    }
+  }
+
+  /**
+    * Retrieve all existing data stores
+    */
+  get("/ds/?") {
+    try {
+      getPersistedDataStores
+    } catch {
+      case e: Exception => handleError(s"Error reading data stores:", e)
+    }
+  }
+
+  /**
+    * Deletes for blobs given id
+    */
+  delete("/blob/:alias/:id/?") {
+    val alias = params("alias")
+    blobStores.get(alias) match {
+      case None =>
+        BadRequest(reason = "AccumuloBlobStore is not initialized.")
+      case Some(abs) =>
+        val id = params("id")
+        logger.debug("Attempting to delete: {} from store: {}", id, alias)
+        abs.delete(id)
+        Ok(reason = s"deleted feature: $id")
+    }
+  }
+
+  /**
+    * Get for blobs given id
+    */
+  get("/blob/:alias/:id/?") {
+    val alias = params("alias")
+    blobStores.get(alias) match {
+      case None =>
+        BadRequest(reason = "AccumuloBlobStore is not initialized.")
+      case Some(abs) =>
+        val id = params("id")
+        logger.debug("Attempting to get blob for id: {} from store: {}", id, alias)
+        val (returnBytes, filename) = abs.get(id)
+        if (returnBytes == null) {
+          NotFound(reason = s"Unknown ID $id")
+        } else {
+          contentType = "application/octet-stream"
+          response.setHeader("Content-Disposition", "attachment;filename=" + filename)
```

Comment:
maybe as a follow up ticket, we could add support for return types instead of downloads, so you could use this to display things on a web page. Might not be a use case though.